### PR TITLE
Add option to pass a schema to the Neo4jPGDS

### LINKED ISSUE #574: Allow to pass a schema to Neo4jPGDS
On a large DB the schema computation can take long, it would be useful to be able to pass in a previously computed (sub-)schema.

### PR DESCRIPTION
**Pull request by pstutz**
Fixes #574, result of conversation at https://github.com/opencypher/cypher-for-apache-spark/issues/573

### CODE REVIEW

**Comment by s1ck** [comment]
Didn't we have that option at some point? Interesting :)

**Comment by Mats-SX** [comment]
I added an example and added an API call for input of a schema file path (let the constructor do the IO for the user).

File: spark-cypher/src/main/scala/org/opencypher/spark/api/GraphSources.scala [hunk 1]
```diff
@@ -62,12 +67,20 @@ object CypherGraphSources {
...
     * @param session            CAPS session
     * @return Neo4j Property Graph Data Source
     */
-  def neo4j(config: Neo4jConfig, omitImportFailures: Boolean = false)
+  def neo4j(config: Neo4jConfig, maybeSchema: Option[Schema] = None, omitImportFailures: Boolean = false)
     (implicit session: CAPSSession): Neo4jPropertyGraphDataSource =
-    Neo4jPropertyGraphDataSource(config, omitImportFailures = omitImportFailures)
+    Neo4jPropertyGraphDataSource(config, maybeSchema = maybeSchema, omitImportFailures = omitImportFailures)
+
+  def neo4j(config: Neo4jConfig, schemaFile: String, omitImportFailures: Boolean)
```
[the remark this answers is not in the document]
Comment (Mats-SX) [reply spark-cypher/src/main/scala/org/opencypher/spark/api/GraphSources.scala]: I tried, but it doesn't compile: https://stackoverflow.com/questions/4652095/why-does-the-scala-compiler-disallow-overloaded-methods-with-default-arguments

[the remark this answers is not in the document]
Comment (Mats-SX) [reply spark-cypher/src/main/scala/org/opencypher/spark/api/GraphSources.scala]: I figure that if you're supplying a schema parameter, you're advanced enough to also learn what value to give `omitImportFailures`.

An alternative is to overload this old-school style with two functions; one that takes the boolean and one that doesn't.

**Pull request merged**